<?xml version="1.0"?>
<robot name="my_first_robot">
  <link name="world"/>

  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="2.0" radius="0.035"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="2.0" radius="0.035"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
    </collision>
  </link>

  <link name="link_0">
    <visual>
      <geometry>
        <box size="1 .1 .1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.5 0 0"/>
    </visual>
    <collision>
      <geometry>
        <box size="1 .1 .1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.5 0 0"/>
    </collision>
  </link>

  <link name="link_1">
    <visual>
      <geometry>
        <box size="1 .05 .05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.5 0 0"/>
    </visual>
    <collision>
      <geometry>
        <box size="1 .05 .05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.5 0 0"/>
    </collision>
  </link>

  <link name="tool0"/>

  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
  </joint>

  <joint name="joint0" type="revolute">
    <origin xyz="0.0 0.0 2." rpy="0.0 0.0 0.0"/>
    <axis xyz="0 1 0"/>
    <parent link="base_link"/>
    <child link="link_0"/>
    <limit lower="0.0" upper="6.28" effort="300.0" velocity="1.0"/>
  </joint>

  <joint name="joint1" type="revolute">
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <parent link="link_0"/>
    <child link="link_1"/>
    <origin rpy="0 0 0" xyz="1 0 0"/>
    <limit lower="0.0" upper="6.28" effort="300.0" velocity="1.0"/>
  </joint>

  <joint name="joint_ef" type="fixed">
    <parent link="link_1"/>
    <child link="tool0"/>
    <origin rpy="0 0 0" xyz="1 0 0"/>
  </joint>

  <transmission name="transmission_joint0">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint0">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_joint0">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="transmission_joint1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <ros2_control name="MyFirstRobotHardware" type="system">
    <hardware>
      <plugin>fake_components/GenericSystem</plugin>
      <param name="joint_names">joint0 joint1</param>
      <param name="command_interfaces">position</param>
      <param name="state_interfaces">position velocity effort</param>
    </hardware>
    <joint name="joint0">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="joint1">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
  </ros2_control>
</robot>